src/host-config/index.ts

```diff
@@ -0,0 +1,132 @@
+import { promises as fs } from 'fs';
+import { isRight } from 'fp-ts/lib/Either';
+import Reporter from 'io-ts-reporters';
+
+import { disallowedProxyFields } from './proxy';
+import type { RedsocksConfig, ProxyConfig } from './types';
+import { HostConfiguration, LegacyHostConfiguration } from './types';
+import { readProxy, setProxy } from './proxy';
+import * as config from '../config';
+import * as applicationManager from '../compose/application-manager';
+import { pathOnRoot } from '../lib/host-utils';
+import log from '../lib/supervisor-console';
+import * as updateLock from '../lib/update-lock';
+
+export * from './proxy';
+export * from './types';
+
+const hostnamePath = pathOnRoot('/etc/hostname');
+
+export async function readHostname() {
+	const hostnameData = await fs.readFile(hostnamePath, 'utf-8');
+	return hostnameData.trim();
+}
+
+export async function setHostname(val: string) {
+	let hostname = val;
+	// If hostname is an empty string, return first 7 digits of device uuid
+	if (!val) {
+		const uuid = await config.get('uuid');
+		hostname = uuid?.slice(0, 7) as string;
+	}
+	// Changing the hostname on config.json will trigger
+	// the OS config-json service to restart the necessary services
+	// so the change gets reflected on containers
+	await config.set({ hostname });
+}
+
+export class HostConfig {
```

Comment:
> parse validates an unknown input and returns an object or throws (instead of warn)

I think we still have to log a warning instead of throwing, because we have to allow for an input to be of type LegacyHostConfiguration even if input doesn't decode to HostConfiguration for backwards compatibility.

Otherwise, SGTM, modified [here]() (TODO)